### PR TITLE
Upgrade default encryption to PBEWITHHMACSHA512ANDAES_256.

### LINKED ISSUE #167: Use a stronger default PBE algorithm in Spring Boot 2+
The current default PBE algorithm is PBEWithMD5AndDES.

As far as I can see, this algorithm is not really secure but, because it is the default, many people are likely to use it.

I know this algorithm is the default in Jasypt itself, which requires backwards compatibility to Java 6.

However, Spring Boot 2 requires Java 8 so would it be possible to set a stronger default algorithm for the Spring Boot 2+ releases e.g. PBEWITHHMACSHA512ANDAES_256? This would also mean defaulting jasypt.encryptor.ivGeneratorClassname to org.jasypt.iv.RandomIvGenerator.

Currently, my organisation uses this for encryption/decryption in 100+ applications as well deployment tooling and not having to set the algorithm to something stronger everywhere would be helpful!

I'm happy to a raise a PR if you are happy to accept this feature request. No worries if not!

### PR DESCRIPTION
**Pull request by rupert-madden-abbott**
Resolves #167

### CODE REVIEW

File: jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/properties/JasyptEncryptorConfigurationProperties.java [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+import java.util.List;
+
 import com.ulisesbocchio.jasyptspringboot.EncryptablePropertyFilter;
 import com.ulisesbocchio.jasyptspringboot.util.AsymmetricCryptography.KeyFormat;
 import lombok.Data;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.boot.context.properties.NestedConfigurationProperty;
 
-import java.util.List;
-
```
Comment (j0rdanit0) [inline jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/properties/JasyptEncryptorConfigurationProperties.java]: No need to move the List import.

Comment (ulisesbocchio) [reply jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/properties/JasyptEncryptorConfigurationProperties.java]: Looks good. I'm gonna try to release this upcoming weekend. Can you create a PR for the samples repo too if any of the tests break?

Comment (rupert-madden-abbott) [reply jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/properties/JasyptEncryptorConfigurationProperties.java]: @j0rdanit0 - agreed! Reverted.

Comment (rupert-madden-abbott) [reply jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/properties/JasyptEncryptorConfigurationProperties.java]: @ulisesbocchio yes no problem. I'll have a look at this now and confirm when done.

Comment (rupert-madden-abbott) [reply jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/properties/JasyptEncryptorConfigurationProperties.java]: @ulisesbocchio samples repo is currently using version 2.1.2 of jasypt-spring-boot (Line 19 in parent pom.xml). Are you happy for my PR to bump this to 2.1.3-SNAPSHOT?

Comment (ulisesbocchio) [reply jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/properties/JasyptEncryptorConfigurationProperties.java]: yeah, that's fine. Thanks!

Comment (rupert-madden-abbott) [reply jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/properties/JasyptEncryptorConfigurationProperties.java]: @ulisesbocchio PR is here: https://github.com/ulisesbocchio/jasypt-spring-boot-samples/pull/9

**Pull request merged**